### PR TITLE
Switch from JSHint to ESLint

### PR DESCRIPTION
**Pull request by mattmahn**
### Summary

We're switching to ESLint because TheFourFifths/consus-client will be using React/JSX, and JSHint does not support that. To make things simpler, we'll use the same linter (ESLint) across all projects.
### Checklist
- [x] Did you run `npm test`?
- [x] Did you update/add documentation for new methods or changed functionality?

Finally, is it ready to be reviewed and merged: YES :white_check_mark:
### How to Review
1. Verify that ESLint is being used by looking in `package.json`/`Gruntfile.js`
2. Run `grunt lint`
### Links
- [TFF-51](https://msoese.atlassian.net/browse/TFF-51)

### CODE REVIEW

File: test/unit/lib/dispatcher.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ describe('Dispatcher', () => {
             let spyA = sinon.spy();
             let spyB = sinon.spy();
             let tokenA = Dispatcher.register(spyA);
+            /* eslint-disable no-unused-vars */
```
Comment (gearedmonkey) [inline test/unit/lib/dispatcher.js]: Why was this added?

Comment (trevinhofmann) [reply test/unit/lib/dispatcher.js]: I'm assuming it was added so the linter would ignore the unused `tokenB` variable.

I would probably just remove the variable, changing the line from `let tokenB = Dispatcher.register(spyB);` to `Dispatcher.register(spyB);`.

**Review by gearedmonkey — APPROVED** [review]

**Comment by trevinhofmann** [comment]
> Run `grunt lint`

Or `npm run lint` to use the local grunt-cli.

File: test/unit/lib/dispatcher.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ describe('Dispatcher', () => {
             let spyA = sinon.spy();
             let spyB = sinon.spy();
             let tokenA = Dispatcher.register(spyA);
-            let tokenB = Dispatcher.register(spyB);
+            Dispatcher.register(spyB);
```
Comment (trevinhofmann) [inline test/unit/lib/dispatcher.js:55]: :+1:

**Review by trevinhofmann — APPROVED** [review]
Looks good to me.
